packages/discord.js/src/util/Channels.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+const { ChannelType } = require('discord-api-types/v10');
+const { ThreadChannelTypes } = require('../util/Constants');
+
+let CategoryChannel;
+let DMChannel;
+let NewsChannel;
+let StageChannel;
+let TextChannel;
+let ThreadChannel;
+let VoiceChannel;
+let DirectoryChannel;
+
+/**
+ * Indicates whether this channel is a {@link ThreadChannel}
+ * @param {Channel} channel The channel to check against
+ * @returns {boolean}
+ */
+function isThread(channel) {
+  return ThreadChannelTypes.includes(channel.type);
+}
+
+/**
+ * Indicates whether this channel is {@link TextBasedChannels text-based}.
+ * @param {Channel} channel The channel to check against
+ * @returns {boolean}
+ */
+function isTextBased(channel) {
+  return 'messages' in channel;
+}
+
+/**
+ * Indicates whether this channel is DM-based (either a {@link DMChannel} or a {@link PartialGroupDMChannel}).
+ * @param {Channel} channel The channel to check against
+ * @returns {boolean}
+ */
+function isDMBased(channel) {
+  return [ChannelType.DM, ChannelType.GroupDM].includes(channel.type);
+}
+
+/**
+ * Indicates whether this channel is {@link BaseGuildVoiceChannel voice-based}.
+ * @param {Channel} channel The channel to check against
+ * @returns {boolean}
+ */
+function isVoiceBased(channel) {
+  return 'bitrate' in channel;
+}
+
+// eslint-disable-next-line valid-jsdoc
+/**
+ * @private
+ */
+function createChannel(client, data, guild, { allowUnknownGuild, fromInteraction } = {}) {
+  CategoryChannel ??= require('./CategoryChannel');
+  DMChannel ??= require('./DMChannel');
+  NewsChannel ??= require('./NewsChannel');
+  StageChannel ??= require('./StageChannel');
+  TextChannel ??= require('./TextChannel');
+  ThreadChannel ??= require('./ThreadChannel');
+  VoiceChannel ??= require('./VoiceChannel');
+  DirectoryChannel ??= require('./DirectoryChannel');
```

Comment:
This should use the `lazy` utility, as well as use on demand, instead of always doing those operations on every run.